pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/ZKMetadataUtils.java

```diff
@@ -66,9 +88,16 @@ private static void updateSegmentZKMetadata(String tableNameWithType, SegmentZKM
     }
 
     if (segmentMetadata.getTimeInterval() != null) {
-      segmentZKMetadata.setStartTime(segmentMetadata.getStartTime());
-      segmentZKMetadata.setEndTime(segmentMetadata.getEndTime());
-      segmentZKMetadata.setTimeUnit(segmentMetadata.getTimeUnit());
+      long startTimeMs = TimeUnit.MILLISECONDS.convert(segmentMetadata.getStartTime(), segmentMetadata.getTimeUnit());
```

Comment:
(minor) start/end time can be easily read from the time interval without converting again

Reply:
https://github.com/apache/pinot/pull/9413#discussion_r981965111